naming/src/main/java/com/alibaba/nacos/naming/push/ClientInfo.java

```diff
@@ -27,161 +26,80 @@
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(ClientTypeDescription.DNSF_CLIENT)) {
-            type = ClientType.DNS;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(ClientTypeDescription.C_CLIENT)) {
-            type = ClientType.C;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(ClientTypeDescription.SDK_CLIENT)) {
-            type = ClientType.JAVA_SDK;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(UtilsAndCommons.NACOS_SERVER_HEADER)) {
-            type = ClientType.NACOS_SERVER;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(ClientTypeDescription.NGINX_CLIENT)) {
-            type = ClientType.TENGINE;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(ClientTypeDescription.CPP_CLIENT)) {
-            type = ClientType.C;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
-        
-        if (versionStr.startsWith(ClientTypeDescription.GO_CLIENT)) {
-            type = ClientType.GO;
-            
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
-            
-            return;
-        }
 
-        if (versionStr.startsWith(ClientTypeDescription.CSHARP_CLIENT)) {
-            type = ClientType.CSHARP;
+    public Version version;
 
-            versionStr = versionStr.substring(versionStr.indexOf(":v") + 2, versionStr.length());
-            version = VersionUtil.parseVersion(versionStr);
+    public ClientType type;
 
+    public ClientInfo(String userAgent) {
+        String versionStr = StringUtils.isEmpty(userAgent) ? StringUtils.EMPTY : userAgent;
+        if (com.alibaba.nacos.common.utils.StringUtils.isBlank(versionStr)) {
```

Comment:
If there use `com.alibaba.nacos.common.utils.StringUtils`, why not replace the import ?